GitUI/CommandsDialogs/SubmodulesDialog/FormAddSubmodule.cs

```diff
@@ -70,20 +72,19 @@ private void DirectorySelectedIndexChanged(object sender, EventArgs e)
 
         private void BranchDropDown(object sender, EventArgs e)
         {
-            var module = new GitModule(Directory.Text);
-
-            var heads = new List<IGitRef>
-            {
-                GitRef.NoHead(module)
-            };
-
-            if (module.IsValidGitWorkingDir())
-            {
-                heads.AddRange(module.GetRefs(false));
-            }
-
-            Branch.DisplayMember = nameof(IGitRef.Name);
-            Branch.DataSource = heads;
+            var gitExecutable = new Executable(() => AppSettings.GitCommand);
+            var heads = gitExecutable.GetOutput(GitCommandHelpers.LsRemotesCmd(url: Directory.Text));
```

Comment:
Why is the command executed outside the module context? I.e. not via GitModule instance?

I think a correct approach would be to construct a module here:
```cs
var submodule = new GitModule(Directory.Text);
var heads = submodule.GitExecutable.GetOutput(GitCommandHelpers.LsRemotesCmd(submodule.WorkingDirectory));
```
or even better encapsulate the functionality int `GitModule`, so we have here:
```cs
var submodule = new GitModule(Directory.Text);
var heads = submodule.GetLsRemotes();
```

Reply:
That's why it worked for local directories only but not for SSH and HTTP[S] links.
`git ls-remotes <url>` is completely independent from a local repo clone.

Reply:
Right, thank you.
It would be good to add comments explaining this.